app/controllers/posts_controller.rb

```diff
@@ -2,9 +2,81 @@ class PostsController < ApplicationController
   def index
     @posts = Post.where(author_id: params[:user_id])
     @user_data = User.find(params[:user_id])
+    @current_user_id = current_user.id
   end
 
   def show
     @post = Post.where(author_id: params[:user_id]).where(id: params[:id])[0]
+    @current_user_id = current_user.id
+  end
+
+  def new
+    new_post = Post.new
+    respond_to do |format|
+      format.html { render :new, locals: { post: new_post } }
+    end
+  end
+
+  def create
+    new_post = Post.new(params.require(:post).permit(:title, :text))
+    new_post.author_id = current_user.id
+    new_post.save
+
+    respond_to do |format|
+      format.html do
+        if new_post.save
+          # Success message
+          flash.now[:success] = 'Post created successfully'
+          redirect_to user_posts_path(current_user.id)
+        else
+          # Error message
+          flash.now[:error] = 'Error: Post creation failed'
+          render :new, locals: { post: new_post }
+        end
+      end
+    end
+  end
```

Comment:
## Optional
- Please always remember to use strong parameters:  Here is the [link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to this lesson